### ISSUE TITLE
S3 bucket mirroring keeps failing for certain 0-byte file

### ISSUE THREAD

**Issue by jwadolowski** [minio/mc#2384]
I have quite large S3 bucket (millions of objects, total size: ~300-500 GB) that I need to mirror on daily basis. The structure of this bucket is almost completely flat and looks as follows:

```
$ mc ls s3/xyz-datastore | head
[2018-01-21 17:54:30 CET]  35KiB 0000-54d3812a3722d84c2a74c40a77882635e94b
[2018-01-21 17:54:30 CET] 177KiB 0000-6a9c5db3052a0290624fc0d563f88829f0df
[2018-01-21 17:54:30 CET] 127KiB 0000-81b188cc06804d82fcaad6b7bb1eeda344aa
[2018-01-21 17:54:29 CET] 140KiB 0000-c89b6bd516a0f90dc3137abc83b21e811d06
[2018-01-21 17:54:30 CET] 107KiB 0000-d001b89f8b8539808a62cd7cfe9422253d42
[2018-01-21 17:54:30 CET] 914KiB 0000-d682b513867ad0e82b93b01bddb27ef2087f
[2018-01-21 17:54:30 CET] 681KiB 0000-e0de3cb6cee472850fd4088cebdd5a8548f2
[2018-01-21 17:54:30 CET]  53KiB 0000-fca36a7667c30d99d36c2c0e0e166311bd4f
[2018-01-21 17:53:59 CET] 145KiB 0001-01c2ebb3b63712ff563794a474ba6ad8546f
[2018-01-21 17:53:59 CET]  29KiB 0001-055618208da2c481aafdb940c0449661c4a9
...
```

The only exception is `META` directory that contains single 0-byte file:

```
$ mc ls s3/xyz-datastore/META
[2018-02-06 07:02:42 CET]     0B repository-78aba510-8067-4a15-8200-4777fbda2570
```

## Expected behaviour

`mc mirror` copies all the files w/o errors

## Actual behaviour

`mc mirror` keeps complaining about single file no matter how many times I run the script.

```
$ su - some-non-root-user -c '/path/to/mc mirror --json --overwrite --remove  s3/xyz-datastore s3/xyz-daily-datastore-backup-05 2>&1'
...
{"status":"success","source":"s3/xyz/9ce0-4e34240f674bc72680f8b843b1457383161a","target":"s3/xyz-daily-datastore-backup-05/9ce0-4e34240f674bc72680f8b843b1457383161a","size":263965,"totalCount":81,"totalSize":31696669}
{"status":"success","source":"s3/xyz/9e45-e0f7b59920d83c2d804380e30d480b35cbab","target":"s3/xyz-daily-datastore-backup-05/9e45-e0f7b59920d83c2d804380e30d480b35cbab","size":17058,"totalCount":82,"totalSize":31713727}
{"status":"success","source":"s3/xyz/META/repository-78aba510-8067-4a15-8200-4777fbda2570","target":"s3/xyz-daily-datastore-backup-05/META/repository-78aba510-8067-4a15-8200-4777fbda2570","size":0,"totalCount":83,"totalSize":31713727}
{"status":"error","error":{"message":"Failed to copy `https://s3.amazonaws.com/xyz/META/repository-78aba510-8067-4a15-8200-4777fbda2570`.","cause":{"message":"The XML you provided was not well-formed or did not validate against our published schema","error":{"Code":"MalformedXML","Message":"The XML you provided was not well-formed or did not validate against our published schema","BucketName":"","Key":"","RequestID":"FA00D21C64471588","HostID":"ltMeUTE0/QrxUJ0hIbRaGvZzlnCKhWfjMYIGKct9jkQaBtxxhNPcIUdInVYMLJUCnu2IJpwtbx0=","Region":""}},"type":"error","sysinfo":{"host.arch":"amd64","host.cpus":"4","host.lang":"go1.9.1","host.name":"xyz.example.com","host.os":"linux","mem.heap.total":"9.7MB","mem.heap.used":"3.6MB","mem.total":"14MB","mem.used":"3.6MB"}}}
{"status":"success","source":"s3/xyz/a1d2-785fbd386e725b3f0f83402916b4479a43cc","target":"s3/xyz-daily-datastore-backup-05/a1d2-785fbd386e725b3f0f83402916b4479a43cc","size":345703,"totalCount":84,"totalSize":32059430}
{"status":"success","source":"s3/xyz/a300-7e7000f15344797d7b5e8ccf9d9316d2f7d5","target":"s3/xyz-daily-datastore-backup-05/a300-7e7000f15344797d7b5e8ccf9d9316d2f7d5","size":734683,"totalCount":85,"totalSize":32794113}
{"status":"success","source":"s3/xyz/a3a2-870eb91222b3500530bea1fd5a36083af2ab","target":"s3/xyz-daily-datastore-backup-05/a3a2-870eb91222b3500530bea1fd5a36083af2ab","size":113024,"totalCount":86,"totalSize":32907137}
{"status":"success","source":"s3/xyz/a3b4-06abcb8bf401dae34ffcd58bdbd676fb7e70","target":"s3/xyz-daily-datastore-backup-05/a3b4-06abcb8bf401dae34ffcd58bdbd676fb7e70","size":2097152,"totalCount":87,"totalSize":35004289}
...
```

2nd attempt

```
$ su - some-non-root-user -c '/path/to/mc mirror --json --overwrite --remove  s3/xyz-datastore s3/xyz-daily-datastore-backup-05 2>&1'
{"status":"success","source":"s3/xyz-datastore/1d91-292c7763ad88d811970a8351ecabc1de980c","target":"s3/xyz-daily-datastore-backup-05/1d91-292c7763ad88d811970a8351ecabc1de980c","size":40156,"totalCount":1,"totalSize":40156}
{"status":"success","source":"s3/xyz-datastore/25a9-3f291ea810855efe3d16bd73a5a36f9364ff","target":"s3/xyz-daily-datastore-backup-05/25a9-3f291ea810855efe3d16bd73a5a36f9364ff","size":157450,"totalCount":2,"totalSize":197606}
{"status":"success","source":"s3/xyz-datastore/META/repository-78aba510-8067-4a15-8200-4777fbda2570","target":"s3/xyz-daily-datastore-backup-05/META/repository-78aba510-8067-4a15-8200-4777fbda2570","size":0,"totalCount":3,"totalSize":197606}
{"status":"success","source":"s3/xyz-datastore/a057-18582ec1175ac7a56f8d5f3f9ef9847c847b","target":"s3/xyz-daily-datastore-backup-05/a057-18582ec1175ac7a56f8d5f3f9ef9847c847b","size":295307,"totalCount":4,"totalSize":492913}
{"status":"error","error":{"message":"Failed to copy `https://s3.amazonaws.com/xyz-datastore/META/repository-78aba510-8067-4a15-8200-4777fbda2570`.","cause":{"message":"The XML you provided was not well-formed or did not validate against our published schema","error":{"Code":"MalformedXML","Message":"The XML you provided was not well-formed or did not validate against our published schema","BucketName":"","Key":"","RequestID":"F0250229300CD087","HostID":"XFqT+aphGWR96E4sYChkD+b4jQaiY26uZ5Hw7aJPsUZqqpTaPO0ZSxv8EUb66BG2aa6YaMa8Zq0=","Region":""}},"type":"error","sysinfo":{"host.arch":"amd64","host.cpus":"4","host.lang":"go1.9.1","host.name":"xyz.example.com","host.os":"linux","mem.heap.total":"8.6MB","mem.heap.used":"2.7MB","mem.total":"13MB","mem.used":"2.7MB"}}}
```

## Steps to reproduce the behaviour

```
$ su - some-non-root-user -c '/path/to/mc mirror --json --overwrite --remove  s3/xyz-datastore s3/xyz-daily-datastore-backup-05 2>&1'
```

## mc version

```
Version: 2017-12-12T01:08:02Z
Release-tag: RELEASE.2017-12-12T01-08-02Z
Commit-id: 4cb6e3534286819d3e11db7c3659164655020bd3
```

## System information

RHEL 7.4 x64

**Comment by harshavardhana** [comment]
@kannappanr is this fixed?

**Comment by kannappanr** [comment]
@harshavardhana The fix was in minio-go and minio-go needs to be vendorized in mc. Then we can close this issue

**Comment by kannappanr** [comment]
Fixed by #2399
